src/python/datatype.rs

```diff
@@ -104,6 +104,16 @@ impl PyDataType {
         Ok(DataType::Date.into())
     }
 
+    #[staticmethod]
+    pub fn list(name: &str, data_type: Self) -> PyResult<Self> {
+        Ok(DataType::List(Box::new(Field::new(name, data_type.dtype))).into())
+    }
+
+    #[staticmethod]
+    pub fn fixed_size_list(name: &str, data_type: Self, size: usize) -> PyResult<Self> {
```

Comment:
I would avoid using `usize` in the pyo3 layer since if the number passed in is a negative or float from python it will cause a panic. I typically end up using a i64 and check at the python layer that it is an integer

Reply:
I'll add the Python-level type and value validation! As for not taking a `usize` at the pyo3 layer, I can change it to an `i64`, although that won't guard against the `float` case. `usize` is the expected type at the Arrow layer when constructing the Arrow type, so we'll need to do that cast at some point; I'll change the pyo3 layer to accept an `i64` and do that cast before handing off to our `DataType::FixedSizeList`.